### PR TITLE
feat: add routing per tabs

### LINKED ISSUE #4: add routing parameter per tab

### PR DESCRIPTION
**Pull request by drendog**
Close #4

### CODE REVIEW

File: src/app/home/home.component.ts [hunk 1]
```diff
@@ -25,6 +32,32 @@ export class HomeComponent {
...
+    }
+  }
+  
+  get isCatalogueLoading(): boolean {
+    return this.catalogueService.isLoading;
+  }
+
+  get tabIndex(): number {
+    if(this.isCatalogueLoading)
+      return -1;
```
Comment (Helias) [inline src/app/home/home.component.ts]:
```suggestion
    if (this.isCatalogueLoading) {
      return -1;
    }
```

File: src/app/home/home.component.ts [hunk 2]
```diff
@@ -25,6 +32,32 @@ export class HomeComponent {
...
+
+    const tabName = this.route.snapshot.paramMap.get('tab');
+    return tabName ? this.catalogueService.confTabPaths.indexOf(`/${tabName}`) : 0;
+  }
+
+  getPathByTab(tab: string): string {
+    return this.catalogueService.CONF.tabs[tab].path;
+  }
```
Comment (Helias) [inline src/app/home/home.component.ts]: this method should be on catalogueService I think

File: src/app/home/home.component.ts [hunk 3]
```diff
@@ -25,6 +32,32 @@ export class HomeComponent {
...
+    const path = `/tab${this.catalogueService.CONF.tabs[tabName].path}`;
+    
+    if (this.location.path() !== path) {
+      this.location.go(path);
+    }
+  }
+  
+  get isCatalogueLoading(): boolean {
+    return this.catalogueService.isLoading;
+  }
```
Comment (Helias) [inline src/app/home/home.component.ts]: is this a bit redundant? you can just get this value from the catalogue service, it's public
creating a function is not good, specially if you use this getter in the web view, this craetes a mechanism that will trigger this getter function everytime the html view is refreshed (and it's refreshed everytime you click on a button or somewhere)

If you use directly catalogueService.isLoading you will save this function call

File: src/app/services/catalogue/catalogue.service.ts [hunk 4]
```diff
@@ -11,6 +11,7 @@ import { Config, Tab } from './catalogue.model';
 export class CatalogueService {
   CONF: Config;
   items$: Record<string, Observable<Repository>> = {};
+  isLoading: boolean = true;
```
Comment (Helias) [inline src/app/services/catalogue/catalogue.service.ts]: the type is inferred, the type declaration here is a bit useless imho

**Comment by Helias** [comment]
@drendog btw good job

File: src/app/home/home.component.ts [hunk 5]
```diff
@@ -25,6 +32,32 @@ export class HomeComponent {
...
+
+  get tabIndex(): number {
+    if(this.isCatalogueLoading)
+      return -1;
+
+    const tabName = this.route.snapshot.paramMap.get('tab');
+    return tabName ? this.catalogueService.confTabPaths.indexOf(`/${tabName}`) : 0;
+  }
```
Comment (Helias) [inline src/app/home/home.component.ts]: not sure if this should be in the home component, I think it is better to move this function in the catalogueService, becauseit can be useful in multiple components

File: src/app/home/home.component.ts [hunk 6]
```diff
@@ -41,21 +41,14 @@ export class HomeComponent {
...
-  
-  get isCatalogueLoading(): boolean {
-    return this.catalogueService.isLoading;
-  }
 
   get tabIndex(): number {
-    if(this.isCatalogueLoading)
-      return -1;
-
     const tabName = this.route.snapshot.paramMap.get('tab');
```
Comment (Helias) [inline src/app/home/home.component.ts]: you could use the route service as dependency of catalogue service and move the entire method into catalogueService

and use catalogueService.tabIndex in the web html view

Comment (drendog) [reply src/app/home/home.component.ts]: i tried, but the route dependency didn't work, `this.route.snapshot.paramMap.get('tab');` will return always `null` in this case, maybe only works only on component class :thinking:

Comment (Helias) [reply src/app/home/home.component.ts]: ok may it's not a good practice but you could pass "this.router" to the method of the service

I got this idea from https://stackoverflow.com/questions/40219790/angular-2-get-routeparams-in-a-service

but I am still not convinced

we could even leave as it is

**Review by Helias — APPROVED** [review]

File: src/app/home/home.component.ts (hunk 6 above)
Comment (drendog) [reply src/app/home/home.component.ts]: solved

File: src/app/home/home.component.ts [hunk 7]
```diff
@@ -11,6 +11,7 @@ import { Location } from '@angular/common';
 @Component({
   selector: 'app-home',
   templateUrl: './home.component.html',
+  providers: [CatalogueService],
```
Comment (Helias) [inline src/app/home/home.component.ts:13]: not bad 👍

**Review by Helias — APPROVED** [review]

File: src/app/home/home.component.ts [hunk 8]
```diff
@@ -4,13 +4,21 @@ import { faTelegram } from '@fortawesome/free-brands-svg-icons';
...
 })
 export class HomeComponent {
-  constructor(public catalogueService: CatalogueService, public cdRef: ChangeDetectorRef) {}
+  constructor(
+    public catalogueService: CatalogueService,
+    public cdRef: ChangeDetectorRef,
+    private route: ActivatedRoute,
```
Comment (Helias) [inline src/app/home/home.component.ts]: is this still necessary?

Comment (drendog) [reply src/app/home/home.component.ts]: no

Comment (Helias) [reply src/app/home/home.component.ts]: than delete it

**Review by Helias — APPROVED** [review]

**Pull request merged**